modules/dnn/src/op_inf_engine.cpp

```diff
@@ -891,6 +891,13 @@ bool InfEngineBackendNet::isInitialized()
 #endif
 }
 
+void InfEngineBackendNet::reset()
+{
+    allBlobs.clear();
+    infRequests.clear();
```

Comment:
What is about multiple async requests with different shapes?

Reply:
That's good point but I think we can cover it separately because even using native Inference Engine API it's tricky. After each network reshaping it's required to load is into the plugin (`LoadNetwork`) then manage asynchronous requests from `ExecutableNetwork::CreateInferRequest`.